src/amo/components/AddonReview.js

```diff
@@ -46,20 +48,24 @@ export class AddonReviewBase extends React.Component {
     const { reviewBody, reviewTitle } = this.state;
     event.preventDefault();
     event.stopPropagation();
+
+    const addonSlug = this.props.review.addonSlug;
+    const apiState = this.props.apiState;
+
     const params = {
+      addonSlug,
+      apiState,
       body: reviewBody,
       title: reviewTitle,
-      addonSlug: this.props.review.addonSlug,
       errorHandler: this.props.errorHandler,
       reviewId: this.props.review.id,
-      apiState: this.props.apiState,
     };
     // TODO: render a progress indicator in the UI.
     // https://github.com/mozilla/addons-frontend/issues/1156
     return this.props.updateReviewText(params)
-      .then(() => {
-        overlayCard.hide();
-      });
+      // This will update the review count on the add-on detail page.
+      .then(() => this.props.refreshAddon({ addonSlug, apiState }))
```

Comment:
This will hold the window open while it makes another API request. Should I close it immediately and let that run in the background? Deferring it to the background means the review count update will be delayed.